packages/expo-dev-client/android/src/androidTest/java/com/expo/modules/devclient/tests/DevClientLoadLocalAppTest.kt

```diff
@@ -0,0 +1,240 @@
+    """.trimIndent()
+
+private const val appURL = "http://localhost:1234"
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+internal class DevClientLoadLocalAppTest : DevLauncherKoinTest() {
+  private val rnHostCreator: ReactNativeHostCreator = { app ->
+    object : ReactNativeHost(app) {
+      override fun getUseDeveloperSupport(): Boolean {
+        return false
+      }
+
+      override fun getPackages(): List<ReactPackage> {
+        return listOf(
+          MainReactPackage(null),
+          DevLauncherPackage(),
+          DevMenuPackage(),
+          object : ReactPackage {
+            override fun createNativeModules(reactContext: ReactApplicationContext) = listOf(DevClientTestExtension(reactContext))
+            override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<View, ReactShadowNode<*>>> = emptyList()
+          })
+      }
+    }.also {
+      reactNativeHostHolder = it
+    }
+  }
+
+  private val koinDeclaration: KoinDeclaration = {
+    declareInDevLauncherScope<DevLauncherAppLoaderFactoryInterface> {
+      object : DevLauncherAppLoaderFactoryInterface, DevLauncherKoinComponent {
+        private val context: Context by inject()
+        private val appHost: ReactNativeHost by inject()
+        private val parsedManifest = StringReader(manifest).use {
+          DevLauncherManifest.fromJson(it)
+        }
+        private val controller: DevLauncherControllerInterface by inject()
+
+        override suspend fun createAppLoader(url: Uri, manifestParser: DevLauncherManifestParser): DevLauncherAppLoader {
+          Truth.assertThat(url).isEqualTo(Uri.parse(appURL))
+          return DevLauncherPublishedAppLoader(
+            parsedManifest,
+            getFileFromAssets(context, "bundled_app.bundle").absolutePath,
+            appHost,
+            context,
+            controller
+          )
+        }
+
+        override fun getManifest(): DevLauncherManifest = parsedManifest
+
+        override fun shouldUseDeveloperSupport(): Boolean = false
+
+        @Throws(IOException::class)
+        fun getFileFromAssets(context: Context, fileName: String): File = File(context.cacheDir, fileName)
+          .also { file ->
+            if (file.exists()) {
+              file.delete()
+            }
+
+            file.outputStream().use { cache ->
+              context.assets.open(fileName).use { inputStream ->
+                inputStream.copyTo(cache)
+              }
+            }
+          }
+      }
+    }
+  }
+
+  @Test
+  fun checks_if_local_app_can_be_open() = DevLauncherBasicScenario(
```

Comment:
```suggestion
  fun checks_if_published_app_can_be_opened() = DevLauncherBasicScenario(
```
Possible rename depending on the answer to the above question